lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -257,10 +256,13 @@ def main():
     required_together = [['s3_key', 's3_bucket'],
                          ['vpc_subnet_ids', 'vpc_security_group_ids']]
 
+    required_if = [['state', 'present', ['runtime', 'handler', 'role']]]
```

Comment:
An update mode is an absolute no-go for me. We have that with rds and we all agree it was (in retrospect) a huge mistake.

I think the current approach that a lot of modules seem to take is to not change things if things aren't passed but it can be difficult to achieve such a model - I guess that's what @s-hertel is going for with #24804. I'm ok with it, for me best practice should be to pass everything that's needed for creation or update for true idempotency, but we don't necessarily need to enforce that best practice.

The third approach would be to have separate modules - one for lambda to manage the lambda configuration, and one for lambda_function that allows for updates of the function only. Obviously the lambda module would continue to be appropriate for most use cases. There are already examples of this kind of thing (e.g. (the now deprecated) ec2_vpc_net vs all the ec2_vpc_net_* submodules)